File: 92backpack/backpack.java
public class Solution {
    /**
     * @param m: An integer m denotes the size of a backpack
     * @param A: Given n items with size A[i]
     * @return: The maximum size
     */
    public int backPack(int m, int[] A) {
        // write your code here
        final int[] dp = new int[m + 1];
        dp[0] = 1;
        Arrays.stream(A).forEach(item -> java.util.stream.IntStream.range(item, m + 1).boxed().sorted(Comparator.reverseOrder()).forEach(size -> dp[size] |= dp[size - item]));
        return Arrays.stream(dp).boxed().collect(java.util.stream.Collectors.toList()).lastIndexOf(1);
    }
}

File: 92backpack/backpack.js
/**
 * @param m: An integer m denotes the size of a backpack
 * @param A: Given n items with size A[i]
 * @return: The maximum size
 */
const backPack = function (m, A) {
    const dp = [1, ...Array(m).fill(0)]
    A.forEach(item => {
        for (let size = m; size > item - 1; --size) dp[size] |= dp[size - item]})
    return dp.lastIndexOf(1)
}
